apis/v1beta1/gateway_types.go

```diff
@@ -69,68 +69,93 @@ type GatewaySpec struct {
 	// logical endpoints that are bound on this Gateway's addresses.
 	// At least one Listener MUST be specified.
 	//
-	// Each listener in a Gateway must have a unique combination of Hostname,
-	// Port, and Protocol.
+	// Each Listener in a set of Listeners (for example, in a single Gateway)
+	// must be _distinct_, in that a traffic flow must be able to be assigned to
+	// exactly one listener. (This section uses "set of Listeners" rather than
+	// "Listeners in a single Gateway" because implementations MAY merge configuration
+	// from multiple Gateways onto a single data plane, and these rules _also_
+	// apply in that case).
```

Comment:
I think this is referring to merging Gateways like the following:

Gateway 1:
- Listener A
- Listener B

Gateway 2:
- Listener C

Resulting merged Gateway:
- Listener A
- Listener B
- Listener C

All of these listeners would still need to be both distinct and isolated, they would just share the same address.

Reply:
Yes, that's the sort of merging I was talking about.

Also, for cascading Listeners, each flow can _individually_ be assigned to a single Listener. This will absolutely need further clarification once we add `GatewayListenerIsolation` as an extended feature.